### ISSUE TITLE
Allow customization of author name's format

### ISSUE THREAD

**Issue by krillin666** [sawhney17/logseq-citation-manager#30]
Currently, the authors name are imported as [[last name]], [[first name]] which creates unnecessary pages and duplicates many authors name since the first or the last name can be a common name.
In the Bibnotes plugin you can specify how the authors name is imported:
![image](https://user-images.githubusercontent.com/60305303/179581714-6e9d8855-4ae6-44b0-be88-fdde347e5cd6.png)

Which looks like this in Logseq:
![image](https://user-images.githubusercontent.com/60305303/179581831-86082b6e-2bc8-4202-ba90-afbd82f6b617.png)

**Comment by sawhney17** [comment]
Can you share your template? The plug-in shouldn't create pages unless you use {authors++}

**Comment by krillin666** [comment]
I’m using `Authors:: {author+} for the author's entry

**Comment by sawhney17** [comment]
Does author++ do a better job? Can you share the entry in the .bib file?

**Comment by jacobtfisher** [comment]
Hi there. Chiming in to say that I also have this issue. I've tried using `{author}` , `{author+}` and `{author++}`, and it creates pages for all of them. I had the thought that it might be a "feature" of Logseq though, since I've noticed that when I use the built-in Zotero integration it also automatically creates pages for each of the authors in the `author::` attribute.

**Comment by sawhney17** [comment]
Logseqs default behavior is now to make properties into pages. Which makes this a little difficult to deal with.

**Comment by jacobtfisher** [comment]
Good point. As far as I can tell, the `authors::` behavior is unique because it doesn't just create property values - it automatically wraps them in square brackets and makes each author an actual page. I assume it's something within the inner workings of the built-in Zotero integration. One thing that I've figured out as a workaround is to put the value for the `authors::` property in quotes so it doesn't create pages. Something like the following:

```
original-title:: Design choices: Mechanism design and platform capitalism
date:: 2021-07-01
authors:: "Viljoen, S., Goldenfein, J., & McGuigan, L."
journal:: Big Data and Society
doi:: 10.1177/20539517211034312
uri:: http://zotero.org/users/5397735/items/LV5TPYGQ
select-link:: zotero://select/library/items/LV5TPYGQ
```

**Comment by krillin666** [comment]
> Logseqs default behavior is now to make properties into pages. Which makes this a little difficult to deal with.
> […](#)
> On 27 Jul 2022 at 3:43 AM +0530, Jacob T. Fisher ***@***.***>, wrote: Hi there. Chiming in to say that I also have this issue. I've tried using {author} , {author+} and {author++}, and it creates pages for all of them. I had the thought that it might be a "feature" of Logseq though, since I've noticed that when I use the built-in Zotero integration it also automatically creates pages for each of the authors in the author:: attribute. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>


Hey @sawhney17 , please see my original post. There I demonstrate how using the Bibnotes plugin in obsidian I get correctly formatted authors names in Logseq.
In the Bibnotes plugin, I specify that authors name follow this format `{first name} {last name}`, so when the article gets imported Bibnotes writes the authors as `authors:: {first name author1} {last name author1}, {first name author2} {last name author2}, etc`

**Comment by sawhney17** [comment]
Cool. Got what you're saying. Will give it a go this week.

**Comment by menelic** [comment]
@sawhney17 did you have any chance to have a look yet? It would really be a boon to usability if this could be made to work as requested by @krillin666

**Comment by krillin666** [comment]
Is there any ETA on this ? :smile:

**Comment by sawhney17** [comment]
flew under my radar, sorry. Will be adding this today

**Comment by menelic** [comment]
Thanks for adding a key functionality for academic an research use cases - It would be great if you could document this as well as the author+, ++ and pdf+ functionality in the readme. Increasingly, users with no knowledge of BibTeX are using it, so there is a need for a more detailed documentation of core functionality, ideally with a few screenshots or gif video.

**Comment by tartakis** [comment]
Any updates on configuring the output of `{author}` as requested by @krillin666 ?

**Comment by alchemiker** [comment]
Hello, I'm here to support this feature request too ✋ I use this plugin now for about a week and absolutely love it, the only thing I dislike about it, that it translates `**Authors**: {author+}` in the template to something like:

> **Authors**: He, Jie,Unser, Sarah,Bruzas, Ian,Cary, ReJeana,Shi, Zhiwei,Mehra, Rajesh,Aron, Kenneth,Sagle, Laura

Which is an ugly acculmulation of first and second names in an almost indistinguishable way. If implementing a customizable name format is too troublesome, please at least include a semicolon `;` as separator between names and maybe improve the spacing.
